collab-folder/src/core/folder.rs

```diff
@@ -206,6 +206,55 @@ impl Folder {
     Some(view)
   }
 
+  pub fn move_nested_view(
+    &self,
+    view_id: &str,
+    new_parent_id: &str,
+    new_prev_id: Option<String>,
+  ) -> Option<Arc<View>> {
+    tracing::debug!("Move nested view: {}", view_id);
+    let view = self.views.get_view(view_id)?;
+    let current_workspace = self.get_current_workspace()?;
+    let current_workspace_id = current_workspace.id.as_str();
+    let parent_id = view.parent_view_id.as_str();
+
+    let new_parent_view = self.views.get_view(new_parent_id);
+    if new_parent_id != current_workspace_id && new_parent_view.is_none() {
```

Comment:
add a comment here. 

This means that the new parent is the workspace if `new_parent_view` is None.

`new_parent_id != current_workspace_id && new_parent_view.is_none()`

I'm not sure if there's a better way to filter this event.

Reply:
Because if can not get parent from views, the parent is wokspace, when this workspace is not equal current workspace, we didnt support move out current workspace yet